powerpc, sparc and many other 32-bit arches need
explicit libatomic for atd::atomic primitives..
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -67,12 +67,18 @@ if(WIN32)
   endif()
 endif()
 
+option(LINK_WITH_ATOMIC "Add 'atomic' if needed" OFF)
+set(atomic_lib "")
+if(LINK_WITH_ATOMIC)
+    set(atomic_lib "atomic")
+endif()
+
 set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
 target_link_libraries(
   ccache_lib
   PRIVATE standard_settings standard_warnings ZSTD::ZSTD
-          Threads::Threads third_party_lib)
+          Threads::Threads third_party_lib ${atomic_lib})
 
 target_include_directories(ccache_lib PRIVATE ${CMAKE_BINARY_DIR} .)
